### PR TITLE
Update CLI Reference with `--site` & `--base` flags

### PR DESCRIPTION
**Pull request by natemoo-re**
#### What kind of changes does this PR include?

- New or updated content

#### Description

- Adds missing `--site` flag to CLI reference, which has been supported for a long time.
- Adds new `--base` flag to CLI reference, which is implemented in https://github.com/withastro/astro/pull/4917

### CODE REVIEW

**Comment by sarah11918** [comment]
LGTM, @natemoo-re! Just to make sure docs are coordinated with the feature release properly, because `--site` already exists, but `--base` is new, and not yet released...

Either this PR needs to wait on the upstream PR https://github.com/withastro/astro/pull/4917 to merge, and we can apply the `merge-on-release` label to remind us to wait. OR

We can use the `<Since /> component to specify the version of Astro that `--base` works with (even if it's a future version) and it's less of a concern if Docs are merged before the feature is released. This would look like:

```
---
setup: |
  import Since from '~/components/Since.astro';
 ---
### `--base`

<Since v="1.4.0" />
```
(Might we want to do this anyway, in case people on older versions try to use `--base` ?)

File: src/pages/en/reference/cli-reference.md [hunk 1]
```diff
@@ -167,6 +168,16 @@ Specifies the path to the config file relative to the project root. Defaults to
 astro --config config/astro.config.mjs dev
 ```
 
+### `--site`
```
Comment (delucis) [inline src/pages/en/reference/cli-reference.md:171]: Do these need to indicate the arguments they get passed like we do for `--root` and `--config`?

```suggestion
### `--site <url>`
```

File: src/pages/en/reference/cli-reference.md [hunk 2]
```diff
@@ -167,6 +168,16 @@ Specifies the path to the config file relative to the project root. Defaults to
...
+
+Configures the [`site`](/en/reference/configuration-reference/#site) for your project. Passing this flag will override the `site` value in your `astro.config.mjs` file, if one exists.
+
+### `--base`
```
Comment (delucis) [inline src/pages/en/reference/cli-reference.md:175]:
```suggestion
### `--base <path>`
```

**Review by delucis — COMMENTED** [review]
Looks good Nate! One comment with how we show these headings. Not entirely happy with my proposed argument names, happy for you to change those if you think something else would be better.

**Comment by delucis** [comment]
(added the flags to the PR title so we can test #1653)

**Pull request merged**